src/applications/yellow-ribbon/containers/SearchForm/index.jsx

```diff
@@ -0,0 +1,123 @@
+// Dependencies.
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import URLSearchParams from 'url-search-params';
+import { map } from 'lodash';
+// Relative imports.
+import STATES from '../../constants/STATES.json';
+import { fetchResultsThunk } from '../../actions';
+
+export class SearchForm extends Component {
+  static propTypes = {
+    // From mapStateToProps.
+    fetching: PropTypes.bool.isRequired,
+    // From mapDispatchToProps.
+    fetchResultsThunk: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+
+    // Derive the current name params.
+    const queryParams = new URLSearchParams(window.location.search);
+
+    // Derive the state values from our query params.
+    const name = queryParams.get('name') || '';
+    const state = queryParams.get('state') || '';
+
+    this.state = {
+      name,
+      state,
```

Comment:
Definitely... I usually use `provincialState` but the data we receive back from the API says `state`, so I didn't want it to get confusing... but it already is confusing so... 🤷‍♂lol

Reply:
Not sure if I should keep it `state` or change it to `provincialState`. I'll need to move these values up into Redux here soon once we get into the comparison page (for the go-back logic). Thoughts/opinion? @ncksllvn

Reply:
I think `state` is okay...I'm kinda wondering if it could be part of another object like `state.formValues.state` or something. But that's not necessary.